File: 2.performance-versus/why-60x/fib.py
def fib(n):
    if n < 2:
        return n
    ans, last = 1, 0  # ans=n=1; last=n=0.
    for _ in range(1, n):
        ans, last = ans + last, ans
    return ans

if __name__ == "__main__":
    import dis
    dis.dis(fib)

File: 2.performance-versus/why-60x/fib.c
long fib(long ord) {
    if (ord < 2) {
        return ord;
    }
    long ans = 1;  // ans=n=1
    long last = 0; // last=n=0
    for (int i=1; i<ord; i++) {
        ans = ans + last;
        last = ans - last;
    }
    return ans;
}
